stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -56,9 +56,10 @@ struct CholeskyPivoted{T,S<:AbstractMatrix} <: Factorization{T}
         new(factors, uplo, piv, rank, tol, info)
     end
 end
-function CholeskyPivoted(A::AbstractMatrix{T}, uplo::AbstractChar, piv::Vector{BlasInt},
-                            rank::BlasInt, tol::Real, info::BlasInt) where T
-    CholeskyPivoted{T,typeof(A)}(A, uplo, piv, rank, tol, info)
+function CholeskyPivoted(A::AbstractMatrix{T}, uplo::AbstractChar, piv::Vector{<:Integer},
+                            rank::Integer, tol::Real, info::Integer) where T
+    CholeskyPivoted{T,typeof(A)}(A, uplo, Vector{BlasInt}(piv), BlasInt(rank), tol,
```

Comment:
Mmm, I missed this, but it's going to make a copy even when not needed. Better use `convert`.